testsuite/MDAnalysisTests/coordinates/test_netcdf.py

```diff
@@ -876,6 +887,133 @@ def test_write_u(self, pos, vel, force, tmpdir, u1, u2):
         u.trajectory.close()
 
 
+class TestNCDFWriterScaleFactors:
+    """Class to check that the NCDF Writer properly handles the
+    application of scale factors"""
+
+    @pytest.fixture()
+    def outfile(self, tmpdir):
+        return str(tmpdir) + 'ncdf-write-scale.ncdf'
+
+    @pytest.fixture()
+    def outfile2(self, tmpdir):
+        return str(tmpdir) + 'ncdf-write-scale2.ncdf'
+
+    @pytest.fixture()
+    def universe(self):
+        return mda.Universe(PRM_NCBOX, TRJ_NCBOX)
+
+    def get_scale_factors(self, ncdfile):
+        """Get a dictionary of scale factors stored in netcdf file"""
+        sfactors = {}
+        with netcdf.netcdf_file(ncdfile, mmap=False) as f:
```

Comment:
Just curious, why is `mmap=False` needed?

Reply:
Honestly the main reason here is "an abundance of caution", I have no idea how the mmap implementation this plays with pytest-xdist so I thought it best to avoid it where possible.

The performance increase is somewhat minimal (given how slow the other parts are), so we're looking at ~ 100 microsecs lost, disabling mmap (at least on my machine it goes from 800 microsec to 900).

I'd probably prefer just being cautious here, but can turn it back on if you don't agree (I'm happy to drop a comment above saying I'm probably being unreasonably cautious 🙃 ).

Reply:
_Maybe_ add a comment anyone else looking at it and wondering. I am not worry about the performance, more along the lines of using tests as implementation examples and then not being sure about why something is done.